### ISSUE TITLE
Video would not render on asus k010

### ISSUE THREAD

**Issue by jozefchutka** [google/ExoPlayer#1114]
I am using exoplayer demo project to test one of my Dash+Widevine video. It does work after modifying WidevineTestMediaDrmCallback.WIDEVINE_GTS_DEFAULT_BASE_URI to my custom value on most of the devices, however there is this strange issue with asus k010 (android 4.4) where I can only hear audio while video would not get rendered. Can you help me find out the issue in the log or point me to something that would help me resolve?

```
01-05 17:56:25.902 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: Could not find method com.google.android.exoplayer.demo.PlayerActivity.requestPermissions, referenced from method com.google.android.exoplayer.demo.PlayerActivity.maybeRequestPermission
01-05 17:56:25.912 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 1428: Lcom/google/android/exoplayer/demo/PlayerActivity;.requestPermissions ([Ljava/lang/String;I)V
01-05 17:56:25.912 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0010
01-05 17:56:25.912 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: Could not find method com.google.android.exoplayer.demo.PlayerActivity.checkSelfPermission, referenced from method com.google.android.exoplayer.demo.PlayerActivity.requiresPermission
01-05 17:56:25.912 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 1395: Lcom/google/android/exoplayer/demo/PlayerActivity;.checkSelfPermission (Ljava/lang/String;)I
01-05 17:56:25.912 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x000e
01-05 17:56:25.942 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.view.accessibility.CaptioningManager$CaptionStyle.hasForegroundColor, referenced from method com.google.android.exoplayer.text.CaptionStyleCompat.createFromCaptionStyleV21
01-05 17:56:25.952 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 286: Landroid/view/accessibility/CaptioningManager$CaptionStyle;.hasForegroundColor ()Z
01-05 17:56:25.952 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0002
01-05 17:56:25.952 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: DexOpt: couldn't find field Landroid/view/accessibility/CaptioningManager$CaptionStyle;.windowColor
01-05 17:56:25.952 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: DexOpt: unable to optimize instance field ref 0x002c at 0x18 in Lcom/google/android/exoplayer/text/CaptionStyleCompat;.createFromCaptionStyleV21
01-05 17:56:26.062 25139-25139/com.google.android.exoplayer.demo I/ExoPlayerImpl: Init 1.5.3
01-05 17:56:26.072 25139-25139/com.google.android.exoplayer.demo D/EventLogger: start [0]
01-05 17:56:26.072 25139-25139/com.google.android.exoplayer.demo D/EventLogger: state [0.00, false, P]
01-05 17:56:26.082 25139-25139/com.google.android.exoplayer.demo D/EventLogger: state [0.01, true, P]
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.MediaCodec.releaseOutputBuffer, referenced from method com.google.android.exoplayer.MediaCodecVideoTrackRenderer.renderOutputBufferV21
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 82: Landroid/media/MediaCodec;.releaseOutputBuffer (IJ)V
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0008
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo E/dalvikvm: Could not find class 'android.media.PlaybackParams', referenced from method com.google.android.exoplayer.MediaCodecAudioTrackRenderer.handleMessage
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve check-cast 50 (Landroid/media/PlaybackParams;) in Lcom/google/android/exoplayer/MediaCodecAudioTrackRenderer;
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x1f at 0x0015
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.AudioTrack.write, referenced from method com.google.android.exoplayer.audio.AudioTrack.writeNonBlockingV21
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 65: Landroid/media/AudioTrack;.write (Ljava/nio/ByteBuffer;II)I
01-05 17:56:27.162 25139-25139/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0001
01-05 17:56:27.172 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to find class referenced in signature (Landroid/media/PlaybackParams;)
01-05 17:56:27.172 25139-25139/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to find class referenced in signature (Landroid/media/PlaybackParams;)
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.view.Display.getMode, referenced from method com.google.android.exoplayer.chunk.VideoFormatSelectorUtil.getDisplaySizeV23
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 252: Landroid/view/Display;.getMode ()Landroid/view/Display$Mode;
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0000
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.MediaCodecInfo$CodecCapabilities.getVideoCapabilities, referenced from method com.google.android.exoplayer.MediaCodecUtil.isSizeAndRateSupportedV21
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 87: Landroid/media/MediaCodecInfo$CodecCapabilities;.getVideoCapabilities ()Landroid/media/MediaCodecInfo$VideoCapabilities;
01-05 17:56:27.182 25139-25584/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0019
01-05 17:56:27.232 25139-25584/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
01-05 17:56:27.282 25139-25584/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
01-05 17:56:27.302 25139-25139/com.google.android.exoplayer.demo D/EventLogger: availableRange [true, 0, 148000000]
01-05 17:56:27.312 25139-25139/com.google.android.exoplayer.demo D/EventLogger: availableRange [true, 0, 147989333]
01-05 17:56:27.332 25139-25139/com.google.android.exoplayer.demo D/EventLogger: state [1.25, true, B]
01-05 17:56:27.392 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.32, 877, 0.02, 292333]
01-05 17:56:27.402 25139-25139/com.google.android.exoplayer.demo D/EventLogger: videoFormat [1.33, 4, 1]
01-05 17:56:27.432 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.36, 804, 0.00, 292333]
01-05 17:56:27.632 25139-25625/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
01-05 17:56:27.672 25139-25625/com.google.android.exoplayer.demo E/ACodec: [OMX.Intel.hw_vd.h264.secure] storeMetaDataInBuffers failed w/ err -2147483648
01-05 17:56:27.692 25139-25139/com.google.android.exoplayer.demo D/EventLogger: decoderInitialized [1.62, OMX.Intel.hw_vd.h264.secure]
01-05 17:56:27.702 25139-25139/com.google.android.exoplayer.demo D/EventLogger: audioFormat [1.63, 9, 2]
01-05 17:56:27.712 25139-25630/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
01-05 17:56:27.722 25139-25139/com.google.android.exoplayer.demo D/EventLogger: decoderInitialized [1.65, OMX.Intel.aac.decoder]
01-05 17:56:27.952 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.88, 536716, 0.49, 8762710]
01-05 17:56:27.962 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.88, 44, 0.00, 8762710]
01-05 17:56:28.262 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.19, 120896, 0.19, 8762710]
01-05 17:56:28.332 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.26, 390187, 0.09, 8762710]
01-05 17:56:28.512 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.43, 52033, 0.08, 8762710]
01-05 17:56:28.562 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.49, 885, 0.01, 8762710]
01-05 17:56:28.892 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.82, 277405, 0.21, 10418967]
01-05 17:56:29.512 25139-25139/com.google.android.exoplayer.demo D/EventLogger: videoFormat [3.43, 7, 3]
01-05 17:56:31.172 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [5.10, 3420920, 2.29, 11976962]
01-05 17:56:31.332 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [5.26, 52033, 0.10, 11976962]
01-05 17:56:33.672 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [7.60, 3482785, 2.33, 11973476]
01-05 17:56:34.072 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [8.00, 469366, 0.33, 11973476]
01-05 17:56:36.142 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [10.07, 3636415, 2.07, 14033439]
01-05 17:56:36.402 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [10.32, 212075, 0.20, 14033439]
01-05 17:56:38.622 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [12.54, 4121888, 2.22, 14833605]
01-05 17:56:38.732 25139-25613/com.google.android.exoplayer.demo D/dalvikvm: GC_FOR_ALLOC freed 1369K, 17% free 7266K/8676K, paused 8ms, total 8ms
01-05 17:56:38.992 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [12.92, 110182, 0.31, 14833605]
01-05 17:56:41.182 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [15.11, 3225422, 2.19, 11787746]
01-05 17:56:41.642 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [15.57, 672898, 0.38, 11787746]
01-05 17:56:43.612 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [17.53, 3159059, 1.97, 12841703]
01-05 17:56:43.932 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [17.86, 177808, 0.25, 12841703]
01-05 17:56:46.202 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [20.13, 3280706, 2.27, 11551781]
01-05 17:56:46.442 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [20.37, 182005, 0.15, 11551781]
01-05 17:56:48.732 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [22.66, 3403554, 2.29, 11895339]
01-05 17:56:49.052 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [22.98, 108988, 0.25, 11895339]
01-05 17:56:51.062 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [24.99, 2339928, 2.01, 9331716]
01-05 17:56:51.492 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [25.42, 400837, 0.34, 9331716]
01-05 17:56:53.632 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [27.55, 3612962, 2.13, 13538030]
01-05 17:56:53.842 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [27.76, 183228, 0.15, 13538030]
01-05 17:56:55.972 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [29.90, 2792717, 2.13, 10484155]
01-05 17:56:56.192 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [30.11, 141357, 0.16, 10484155]
01-05 17:56:58.402 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [32.33, 3162816, 2.22, 11402671]
01-05 17:56:58.712 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [32.64, 199721, 0.23, 11402671]
01-05 17:57:00.842 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [34.76, 2868213, 2.13, 10787825]
01-05 17:57:01.142 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [35.06, 213272, 0.23, 10787825]
01-05 17:57:03.312 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [37.24, 3618006, 2.17, 13338270]
01-05 17:57:03.592 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [37.51, 170412, 0.20, 13338270]
01-05 17:57:05.802 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [39.73, 3184691, 2.22, 11450574]
01-05 17:57:06.082 25139-25139/com.google.android.exoplayer.demo D/EventLogger: bandwidth [40.01, 240400, 0.21, 11450574]
```

My best guess is that the device is missing some kind of capability to decode the stream. Thanks for any suggestions

**Comment by ojw28** [comment]
- Do the bundled clear samples play properly in the demo app? What about the bundled encrypted ones?
- Can you test an equivalent clear sample to the one you're having difficulty with. Does that one work?
- When playing the affected stream in the demo app, if you bring up the controls, is there a Video button at the top listing some available video streams, or are the Audio and Logging buttons the only ones that are present?

**Comment by jozefchutka** [comment]
@ojw28 , thanks for a quick response, I tried most of the bundled dash/widevine streams and I can observe the video is not rendered for any H265 stream (having h265 in title). Either clear or secured, same problem just like my stream. There is no error displayed. And as you say there is no Video button at the top listing when this is happening. 

Any idea how to proceed from here?

There are also problems with other bundled streams but these are not my concern:
- WV: Secure video path required (WebM,VP9) - Unable to instantiate decoder CIPVorbisDecoder error
- Super speed (PlayReady) - device does not support DRM scheme error

**Comment by ojw28** [comment]
- H265 is only supported on Android 5.0 and above. See [here](http://developer.android.com/guide/appendix/media-formats.html).
- The CIPVorbisDecoder issue is fixed in the `dev` branch by https://github.com/google/ExoPlayer/commit/9bcd1069b110fc7158f630905d93ecbe17da5a1f.
- Most devices that are not AndroidTVs do not support PlayReady.

I think you'll need to provide a sample stream for us to debug this further. If you're using H265 then it would be working as intended that you don't get video due to the first point above. Please also try and play an equivalent clear stream; which will separate out DRM issues from anything else.

**Comment by jozefchutka** [comment]
I am using master branch ( https://github.com/google/ExoPlayer/commit/c9caaad50c4148a8c369816c100c170436a3d4d6 ) for my tests, as I can live without PR and vorbis. I have found another android 4.4 device that is samsung SM-P600 and just like you say it would not play H265, but it would play my stream (which is avc btw.). Although, here the video button is present even for H265 even when stream video is not rendered.

So basically, I have samsung SM-P600 playing my stream, while asus k010 not, while both are android 4.4. First I tought the avc coded in .mpd might be wrong, but obviously it can not be hevc as that would not work even on samsung. I am checking if I can expose the mpd with you for check and also clear version of the same.

**Comment by ojw28** [comment]
It's possible the device just isn't capable. As above, we can't really help more until you (a) test an equivalent clear stream to properly determine whether it's a DRM issue or just an issue with the content, and (b) provide a sample stream once this is done.

**Comment by jozefchutka** [comment]
@ojw28 I finally managed to get available streams isolating the issue with asus. There are these two manifests:
- Clear: http://teststreams.livesport-massive.com/out/u/190116010_EN_PERFORM_CAVSREAX_POSTGSW_IV_1_1453370357431_2224_1453381760177.mpd
- Widevine:  http://teststreams.livesport-massive.com/out/u/190116010_EN_PERFORM_CAVSREAX_POSTGSW_IV_1_1453370357431_2224_1453381760177-encrypted.mpd

I can play both on samsung SM-P600 (Android 4.4) but on asus k010 (Android 4.4) I can only play clear one correctly. If I wait long enough (1 minute of buffering) the stream would start to play but audio only. Video remains black screen. Once that is happening I can see Audio, Video and Logging buttons on top. Attaching log:

```
02-04 16:59:02.978 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: Could not find method com.google.android.exoplayer.demo.PlayerActivity.requestPermissions, referenced from method com.google.android.exoplayer.demo.PlayerActivity.maybeRequestPermission
02-04 16:59:02.978 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 1428: Lcom/google/android/exoplayer/demo/PlayerActivity;.requestPermissions ([Ljava/lang/String;I)V
02-04 16:59:02.978 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0010
02-04 16:59:02.988 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: Could not find method com.google.android.exoplayer.demo.PlayerActivity.checkSelfPermission, referenced from method com.google.android.exoplayer.demo.PlayerActivity.requiresPermission
02-04 16:59:02.988 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 1395: Lcom/google/android/exoplayer/demo/PlayerActivity;.checkSelfPermission (Ljava/lang/String;)I
02-04 16:59:02.988 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x000e
02-04 16:59:03.008 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.view.accessibility.CaptioningManager$CaptionStyle.hasForegroundColor, referenced from method com.google.android.exoplayer.text.CaptionStyleCompat.createFromCaptionStyleV21
02-04 16:59:03.008 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 286: Landroid/view/accessibility/CaptioningManager$CaptionStyle;.hasForegroundColor ()Z
02-04 16:59:03.008 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0002
02-04 16:59:03.008 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: DexOpt: couldn't find field Landroid/view/accessibility/CaptioningManager$CaptionStyle;.windowColor
02-04 16:59:03.008 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: DexOpt: unable to optimize instance field ref 0x002c at 0x18 in Lcom/google/android/exoplayer/text/CaptionStyleCompat;.createFromCaptionStyleV21
02-04 16:59:03.098 11510-11510/com.google.android.exoplayer.demo I/ExoPlayerImpl: Init 1.5.3
02-04 16:59:03.108 11510-11510/com.google.android.exoplayer.demo D/EventLogger: start [0]
02-04 16:59:03.108 11510-11510/com.google.android.exoplayer.demo D/EventLogger: state [0.00, false, P]
02-04 16:59:03.118 11510-11510/com.google.android.exoplayer.demo D/EventLogger: state [0.01, true, P]
02-04 16:59:04.058 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.MediaCodec.releaseOutputBuffer, referenced from method com.google.android.exoplayer.MediaCodecVideoTrackRenderer.renderOutputBufferV21
02-04 16:59:04.058 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 82: Landroid/media/MediaCodec;.releaseOutputBuffer (IJ)V
02-04 16:59:04.058 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0008
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo E/dalvikvm: Could not find class 'android.media.PlaybackParams', referenced from method com.google.android.exoplayer.MediaCodecAudioTrackRenderer.handleMessage
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve check-cast 50 (Landroid/media/PlaybackParams;) in Lcom/google/android/exoplayer/MediaCodecAudioTrackRenderer;
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x1f at 0x0015
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.AudioTrack.write, referenced from method com.google.android.exoplayer.audio.AudioTrack.writeNonBlockingV21
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 65: Landroid/media/AudioTrack;.write (Ljava/nio/ByteBuffer;II)I
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0001
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to find class referenced in signature (Landroid/media/PlaybackParams;)
02-04 16:59:04.068 11510-11510/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to find class referenced in signature (Landroid/media/PlaybackParams;)
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.view.Display.getMode, referenced from method com.google.android.exoplayer.chunk.VideoFormatSelectorUtil.getDisplaySizeV23
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 252: Landroid/view/Display;.getMode ()Landroid/view/Display$Mode;
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0000
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo I/dalvikvm: Could not find method android.media.MediaCodecInfo$CodecCapabilities.getVideoCapabilities, referenced from method com.google.android.exoplayer.MediaCodecUtil.isSizeAndRateSupportedV21
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo W/dalvikvm: VFY: unable to resolve virtual method 87: Landroid/media/MediaCodecInfo$CodecCapabilities;.getVideoCapabilities ()Landroid/media/MediaCodecInfo$VideoCapabilities;
02-04 16:59:04.078 11510-12193/com.google.android.exoplayer.demo D/dalvikvm: VFY: replacing opcode 0x6e at 0x0019
02-04 16:59:04.088 11510-12193/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
02-04 16:59:04.118 11510-12193/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
02-04 16:59:04.128 11510-11510/com.google.android.exoplayer.demo D/EventLogger: availableRange [true, 0, 214000000]
02-04 16:59:04.128 11510-11510/com.google.android.exoplayer.demo D/EventLogger: availableRange [true, 0, 213994666]
02-04 16:59:04.148 11510-11510/com.google.android.exoplayer.demo D/EventLogger: state [1.04, true, B]
02-04 16:59:04.218 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.12, 877, 0.02, 318909]
02-04 16:59:04.238 11510-11510/com.google.android.exoplayer.demo D/EventLogger: videoFormat [1.13, 4, 1]
02-04 16:59:04.258 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.15, 804, 0.01, 318909]
02-04 16:59:04.438 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.33, 515752, 0.16, 26280356]
02-04 16:59:04.468 11510-12236/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
02-04 16:59:04.468 11510-12236/com.google.android.exoplayer.demo E/ACodec: [OMX.Intel.hw_vd.h264.secure] storeMetaDataInBuffers failed w/ err -2147483648
02-04 16:59:04.488 11510-11510/com.google.android.exoplayer.demo D/EventLogger: decoderInitialized [1.38, OMX.Intel.hw_vd.h264.secure]
02-04 16:59:04.508 11510-11510/com.google.android.exoplayer.demo D/EventLogger: audioFormat [1.40, 9, 2]
02-04 16:59:04.508 11510-12240/com.google.android.exoplayer.demo I/OMXClient: Using client-side OMX mux.
02-04 16:59:04.528 11510-11510/com.google.android.exoplayer.demo D/EventLogger: decoderInitialized [1.42, OMX.Intel.aac.decoder]
02-04 16:59:04.588 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.48, 52061, 0.06, 26280356]
02-04 16:59:04.878 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.78, 280660, 0.23, 9636394]
02-04 16:59:04.938 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [1.84, 264659, 0.06, 26280356]
02-04 16:59:05.168 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.07, 43366, 0.13, 26280356]
02-04 16:59:05.208 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.11, 9554, 0.04, 9636394]
02-04 16:59:05.638 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [2.53, 289131, 0.31, 9636394]
02-04 16:59:06.228 11510-11510/com.google.android.exoplayer.demo D/EventLogger: videoFormat [3.12, 7, 3]
02-04 16:59:07.858 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [4.76, 2925771, 2.23, 10514900]
02-04 16:59:08.128 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [5.02, 215766, 0.22, 10514900]
02-04 16:59:10.318 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [7.21, 2980501, 2.19, 10897627]
02-04 16:59:10.648 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [7.54, 316642, 0.27, 10897627]
02-04 16:59:12.698 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [9.59, 2891759, 2.05, 11290420]
02-04 16:59:12.888 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [9.78, 126783, 0.13, 11290420]
02-04 16:59:15.088 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [11.98, 3250092, 2.20, 11818516]
02-04 16:59:15.208 11510-12224/com.google.android.exoplayer.demo D/dalvikvm: GC_FOR_ALLOC freed 1340K, 16% free 7294K/8676K, paused 6ms, total 6ms
02-04 16:59:15.458 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [12.35, 294254, 0.30, 11818516]
02-04 16:59:17.528 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [14.42, 3132374, 2.07, 12088273]
02-04 16:59:17.778 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [14.67, 207320, 0.19, 12088273]
02-04 16:59:20.068 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [16.96, 3072703, 2.29, 10724966]
02-04 16:59:20.428 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [17.32, 282848, 0.28, 10724966]
02-04 16:59:22.558 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [19.45, 3124062, 2.13, 11744593]
02-04 16:59:22.778 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [19.67, 190471, 0.16, 11744593]
02-04 16:59:24.998 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [21.89, 3232519, 2.22, 11669743]
02-04 16:59:25.238 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [22.13, 265115, 0.17, 11669743]
02-04 16:59:27.288 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [24.18, 3115925, 2.05, 12165641]
02-04 16:59:27.718 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [24.61, 88341, 0.16, 12165641]
02-04 16:59:29.958 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [26.85, 3505323, 2.24, 12524602]
02-04 16:59:30.478 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [27.37, 220394, 0.21, 12524602]
02-04 16:59:32.668 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [29.56, 3083390, 2.18, 11289299]
02-04 16:59:33.068 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [29.96, 52188, 0.16, 11289299]
02-04 16:59:35.358 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [32.25, 2999782, 2.29, 10484166]
02-04 16:59:35.708 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [32.60, 127426, 0.13, 10484166]
02-04 16:59:38.178 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [35.07, 3051765, 2.48, 9860306]
02-04 16:59:38.878 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [35.77, 275980, 0.43, 9860306]
02-04 16:59:40.948 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [37.85, 2896881, 2.07, 11179473]
02-04 16:59:41.368 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [38.26, 180880, 0.17, 11179473]
02-04 16:59:43.508 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [40.40, 3335442, 2.14, 12480606]
02-04 16:59:43.878 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [40.77, 88685, 0.11, 12480606]
02-04 16:59:46.098 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [42.99, 3692134, 2.22, 13275088]
02-04 16:59:46.588 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [43.48, 88884, 0.23, 13275088]
02-04 16:59:49.138 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [46.03, 4089121, 2.57, 12748623]
02-04 16:59:49.748 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [46.65, 321360, 0.30, 12748623]
02-04 16:59:51.888 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [48.78, 3644382, 2.14, 13630227]
02-04 16:59:52.428 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [49.32, 250208, 0.25, 13630227]
02-04 16:59:54.608 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [51.51, 3627506, 2.19, 13257216]
02-04 16:59:55.018 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [51.92, 240072, 0.17, 13257216]
02-04 16:59:57.158 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [54.05, 3545800, 2.13, 13292596]
02-04 16:59:57.768 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [54.66, 380003, 0.33, 13292596]
02-04 16:59:59.988 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [56.88, 3617406, 2.22, 13012251]
02-04 17:00:00.308 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [57.20, 87437, 0.11, 13012251]
02-04 17:00:02.558 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [59.45, 3763223, 2.25, 13380348]
02-04 17:00:02.868 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [59.76, 52182, 0.10, 13380348]
02-04 17:00:05.048 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [61.94, 3549151, 2.17, 13060353]
02-04 17:00:05.438 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [62.33, 173622, 0.15, 13060353]
02-04 17:00:07.668 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [64.57, 3372984, 2.24, 12057136]
02-04 17:00:08.228 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [65.13, 259341, 0.23, 12057136]
02-04 17:00:10.358 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [67.25, 3404419, 2.12, 12828710]
02-04 17:00:11.188 11510-12224/com.google.android.exoplayer.demo D/dalvikvm: GC_FOR_ALLOC freed 3223K, 29% free 8095K/11364K, paused 7ms, total 7ms
02-04 17:00:11.248 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [68.14, 303423, 0.29, 12828710]
02-04 17:00:13.348 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [70.25, 3671830, 2.10, 13961330]
02-04 17:00:13.688 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [70.58, 52023, 0.07, 13961330]
02-04 17:00:15.948 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [72.85, 3628277, 2.25, 12923515]
02-04 17:00:16.598 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [73.49, 314946, 0.21, 12923515]
02-04 17:00:18.818 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [75.71, 3160710, 2.22, 11384817]
02-04 17:00:19.278 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [76.16, 217227, 0.20, 11384817]
02-04 17:00:21.408 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [78.30, 3186597, 2.14, 11912512]
02-04 17:00:22.028 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [78.92, 282804, 0.31, 11912512]
02-04 17:00:24.058 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [80.95, 2995156, 2.03, 11809387]
02-04 17:00:24.408 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [81.30, 84500, 0.09, 11809387]
02-04 17:00:26.758 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [83.65, 2990597, 2.35, 10180755]
02-04 17:00:27.438 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [84.33, 302256, 0.39, 10180755]
02-04 17:00:29.498 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [86.39, 2973123, 2.06, 11529318]
02-04 17:00:29.878 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [86.77, 82737, 0.10, 11529318]
02-04 17:00:32.018 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [88.91, 3264488, 2.14, 12192298]
02-04 17:00:33.878 11510-11510/com.google.android.exoplayer.demo D/EventLogger: bandwidth [90.77, 2368241, 1.61, 11753057]
02-04 17:00:33.918 11510-11510/com.google.android.exoplayer.demo D/EventLogger: state [90.81, true, R]
```

any hints?

**Comment by jozefchutka** [comment]
extracted the important bits into a new issue https://github.com/google/ExoPlayer/issues/1341
